apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/AssessmentQuestionManualGrading.tsx

```diff
@@ -0,0 +1,189 @@
+
+  const { groupSubmissionMutation, toggleAiGradingModeMutation, ...mutations } =
+    useManualGradingActions({
+      csrfToken,
+      courseInstanceId: courseInstance.id,
+    });
+
+  return (
+    <>
+      {toggleAiGradingModeMutation.isError && (
+        <Alert
+          variant="danger"
+          class="mb-3"
+          dismissible
+          onClose={() => toggleAiGradingModeMutation.reset()}
+        >
+          <strong>Error:</strong> {toggleAiGradingModeMutation.error.message}
+        </Alert>
+      )}
+      <div class="d-flex flex-row justify-content-between align-items-center mb-3 gap-2">
+        <nav aria-label="breadcrumb">
+          <ol class="breadcrumb mb-0">
+            <li class="breadcrumb-item">
+              <a href={`${urlPrefix}/assessment/${assessment.id}/manual_grading`}>Manual grading</a>
+            </li>
+            <li class="breadcrumb-item active" aria-current="page">
+              Question {questionNumber != null ? questionNumber : ''}. {questionTitle}
+            </li>
+          </ol>
+        </nav>
+        {aiGradingEnabled && (
+          <div class="card px-3 py-2 mb-0">
+            <div class="form-check form-switch mb-0">
+              <input
+                class="form-check-input"
+                type="checkbox"
+                role="switch"
+                id="switchCheckDefault"
+                checked={aiGradingMode}
+                disabled={toggleAiGradingModeMutation.isPending}
+                onChange={() =>
+                  toggleAiGradingModeMutation.mutate(undefined, {
+                    onSuccess: () => {
+                      setAiGradingMode((prev) => !prev);
+                    },
+                  })
+                }
+              />
+              <label class="form-check-label" for="switchCheckDefault">
+                <i class="bi bi-stars" />
+                AI grading mode
+              </label>
+            </div>
+          </div>
+        )}
+      </div>
+      <AssessmentQuestionTable
+        hasCourseInstancePermissionEdit={hasCourseInstancePermissionEdit}
+        course={course}
+        courseInstance={courseInstance}
+        csrfToken={csrfToken}
+        instanceQuestionsInfo={instanceQuestionsInfo}
+        urlPrefix={urlPrefix}
+        assessment={assessment}
+        assessmentQuestion={assessmentQuestion}
+        questionQid={questionQid}
+        aiGradingMode={aiGradingMode}
+        rubricData={rubricData}
+        instanceQuestionGroups={instanceQuestionGroups}
+        courseStaff={courseStaff}
+        aiGradingStats={aiGradingStats}
+        mutations={mutations}
+        onSetGroupInfoModalState={setGroupInfoModalState}
+        onShowConflictModal={handleShowConflictModal}
+      />
+
+      <GroupInfoModal
+        modalState={groupInfoModalState}
+        numOpenInstances={numOpenInstances}
+        mutation={groupSubmissionMutation}
+        onHide={() => setGroupInfoModalState(null)}
```

Comment:
Minor node: coupling the "mode" and "visibility" like this means that we lose the exit animation on the modal because it wipes the modal from the DOM immediately (see the `return null` in this component).